Sources/HummingbirdCore/Server/Server.swift

```diff
@@ -114,7 +114,7 @@ public actor HBServer<ChildChannel: HBChildChannel>: Service {
                         await withDiscardingTaskGroup { group in
                             do {
                                 try await asyncChannel.executeThenClose { inbound in
-                                    for try await childChannel in inbound {
+                                    for try await childChannel in inbound.cancelOnGracefulShutdown() {
```

Comment:
You do need to cancel this on gracefulShutdown, or more requests can come in while finishing the processing of the current request. Again we could possibly use `Task.isShuttingDownGracefully`

Reply:
This change is additive, and adds the `cancelOnGracefulShutdown()`. This wasn't removed, contrary to that it was _missing_.

Reply:
Channel is closed in `shutdownGracefully` so the cancelOnGracefulShutdown is not needed